SqlAssignments.sql

```diff
@@ -0,0 +1,1846 @@
+
+
+
+Que 16 -> Select all orders that had amounts that were greater than at least one of the orders from October 6.
+
+Query   -> SELECT onum,amt,odate 
+        -> FROM Orders 
+        -> WHERE amt > (SELECT MIN(amt)
+        ->              FROM Orders
+        ->              WHERE odate ='1990-06-10');
++------+---------+------------+
+| onum | amt     | odate      |
++------+---------+------------+
+| 3002 |  1900.1 | 1990-03-10 |
+| 3005 | 5160.45 | 1990-03-10 |
+| 3008 |    4723 | 1990-05-10 |
+| 3009 | 1713.23 | 1990-04-10 |
+| 3011 | 9891.88 | 1990-06-10 |
++------+---------+------------+
+5 rows in set (0.00 sec)
+
+**************************************************************************************************************************************
+
+
+
+Que 17 ->  Write a query that uses the EXISTS operator to extract all salespeople who have customers with a rating of 300.
+
+Query  ->   SELECT S.snum,S.sname,S.city 
+       ->   FROM salespeople S 
+       ->   WHERE EXISTS( SELECT C.snum                
+                          FROM Customers C               
+                          WHERE C.snum=S.snum AND  C.rating = 300
+                        );
++------+--------+-----------+
+| snum | sname  | city      |
++------+--------+-----------+
+| 1002 | Serres | SanJose   |
+| 1007 | Rifkin | Barcelona |
++------+--------+-----------+
+2 rows in set (0.01 sec)
+
+
+**************************************************************************************************************************************
+
+
+
+Que 18  ->  Find all customers whose cnum is 1000 above the snum of Serres.
+
+Query   -> SELECT C.CNUM,C.CNAME 
+        -> FROM Customers C,Salespeople S 
+        -> WHERE S.SNUM=C.SNUM and C.CNUM>S.SNUM+1000;
++------+---------+
+| CNUM | CNAME   |
++------+---------+
+| 2003 | Liu     |
+| 2004 | Grass   |
+| 2005 | Clemens |
+| 2007 | Pereira |
++------+---------+
+
+
+
+**************************************************************************************************************************************
+
+
+
+Que 19  -> Give the salespeople’s commissions as percentages instead of decimal numbers.
+
+Query   -> SELECT sname,comm*100 AS COMM 
+        -> FROM Salespeople.
+        
++---------+------+
+| SNAME   | COMM |
++---------+------+
+| Peel    | 1200 |
+| Serres  | 1300 |
+| AxelRod | 1000 |
+| Motika  | 1100 |
+| Rifkin  | 1500 |
+| Fran    | 2500 |
++---------+------+
+6 rows in set (0.00 sec)
+ 
+**************************************************************************************************************************************
+ 
+ 
+ 
+Que 20 -> Find the largest order taken by each salesperson on each date, eliminating those Maximum orders, which are less than 3000.
+
+Query  -> SELECT onum,odate,MAX(amt) 
+       -> FROM Orders 
+       -> GROUP BY onum HAVING MAX(amt)>3000;
++------+------------+----------+
```

Comment:
Display salesperson name as well in this query

Reply:
Sure sir